### DISCUSSION TITLE
Can not get Crazyflie to hover in a stable state

### DISCUSSION THREAD

**Discussion by Mitesh-Rengar** [bitcraze/discussions#131] [Q&A]
Hi,

I just bought a Crazyflie drone and completed setting it up. However, I can not get it to hover in a stable state. The Flight data on the Crazyflie  client on the Windows shows that there are some ghost values on the pitch and roll and hence I think the drone is drifting. Am I doing anything wrong, or is this normal?

**Comment by bhavilailogix** [comment]
I am facing the same issue.

**Comment by knmcguire** [comment]
Hi! We need to have more information about your issue:
- Are you using a positioning system? And if so, which one? Lighthouse, Loco, Mocap or a flowdeck?
- Are you able to fly the crazyflie with a controller or the mobile app, as indicated in the getting started guide: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/

**Comment by Mitesh-Rengar** [comment]
Hi Kimberly,
Thanks for your reply.

No, at the moment I am not using any positioning system. I do have a flowdeck, but I have not yet configured it. I just assembled my drone and have been able to fly it using both my mobile and by using a controller through the crazyflie client.

When I try to apply throttle to the drone, the drone does not go straight up, rather it drifts in one direction. Using the telemetry on the crazyflie Windows client, I found out that I am getting some values for Pitch, Roll and Yaw without providing any input.
I have tried to disable the Pitch, Roll and Yaw from the advanced flight control settings, but the drone is still drifting, please see the attached flight data of the drone when kept on a perfectly plain surface. 
![cf telemetry](https://user-images.githubusercontent.com/101316546/178253938-bdf6048e-3abd-456b-9fdd-a852198b6e81.JPG)

**Reply by knmcguire** [reply]
When you start up the drone, was it then on a flat surface at the time? At startup the IMU is calibrated so if it was not flat at the time that might cause problems.

Also, a quadcopter is an unstable system, so without an positioning system there is always drift to the sides, that is not something that will be fully eliminated.

**Reply by Mitesh-Rengar** [reply]
Yes, I put it down on the floor when I first started it, and the IMU gets calibrated properly.  I will try configuring the flowdeck with the drone and see if I can get it to hover.

Thanks for your help.

**Reply by Mitesh-Rengar** [reply]
Hi Kimberly,

I have added the FlowDeck and done a few test flights with it. I can control it well using the controller however it still drifts from time to time. 
The main iussue which I am facing is the drift when I am using a sample python script to control it. Even if I just enter the code to take off, hold for 5 seconds and then land, the drone drifts away from the starting position and is in no way stable. 

Is there anything that you can help me with?

**Comment by knmcguire** [comment]
If you are flying it with a controller + flowdeck, you are either flying it in attitude mode (not with position feedback) or position hold (which uses position feedback). There is an button on the controller that you can use.

But in general, flying with a flow deck is just very much easier to let it do autonomously. Either use the commander panel of the flight tab here (make sure to disconnect your controller before opening the cfclient)
![Screenshot 2022-07-27 180134](https://user-images.githubusercontent.com/11061260/181294761-00760e67-e3d9-44b9-9f8d-880f3da73b80.png)

or  

just try out [the motion commander demo of the cflib.](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/motion_commander_demo.py)
